uktrade/data-hub-frontend
# Reorganize Google Tag Manager snipped and initialization

## Description of change

This PR fixes an issue introduced with #7008 when Google Analytics events have not been sent because the app set default consent to _denied_ for the whole of Data Hub and not just the Export Wins Review part. 

## Test instructions

1. Start a local instance of Data Hub which has the `GOOGLE_TAG_MANAGER_KEY` env var set to an ID of an existing _container_ in [Google Tag Manager](https://tagmanager.google.com/).
2. Go to [Google Tag Assistant] (you might have to install a browser extnsion for it to work)
3. Connect the tool to your local instance by clicking on the _Add domain_ button and typing http://localhost:3000 (or whatever is the address of your Data Hub instance) into _Your website's URL_ field and click the _Connect_ button and then the _Continue_ button, which should open http://localhost:3000 in a new tab
4. In the [Google Tag Assistant] you should see a log of numbered events
5. The event <span>#</span>1 should be _Consent Initialization_, click on it
6. In the event's _Output_ pane, click on the _Consent_ tab
7. It should read "Consent not configured, Default consent state has not been set yet"
9. When you interact with Data Hub in the new tab opened by [Google Tag Assistant]
10. You should see events appearing in the [Google Tag Assistant]
11. Disconnect the tab with Data Hub from [Google Tag Assistant] by clicking on the big cross in the top-left corner
12. Repeat step 3, but this time type `http://localhost:3000/exportwins/review/<token>` into the _Your website's URL_ field
13. Again, you should see a log of numbered events in the [Google Tag Assistant](https://tagassistant.google.com) you should see a log of numbered events
14. This time however, the event <span>#</span>1 should be _Consent Default_ and the _Consent_ tab in the event's _Output_ pane should show a table with all the values in the _On-page Default_ column should be _Denied_ <img width="1275" alt="Screenshot 2024-09-20 at 15 40 57" src="https://github.com/user-attachments/assets/37fd27d3-a9f8-41e2-aa0a-7ad860093a8c">
15. The _Consent Initialization_ should now be event <span>#</span>2 and it's _Content_ tab should show the same table, but with an additional _Current State_ column also with all values _Denied_ <img width="1276" alt="Screenshot 2024-09-20 at 15 43 08" src="https://github.com/user-attachments/assets/daaed2ba-96cf-4481-9090-625fd30910fa">
16. Now in the browser tab with `http://localhost:3000/exportwins/review/<token>` open, click on the _Accept analytics cookies_ button in the cookie banner on top of the page
17. A _Consent update_ event should appear in [Google Tag Assistant] and the table with it's _Consent_ tab should now have the value _Granted_ in the _On-page Update_ column of row _analytics_storage_ only <img width="1279" alt="Screenshot 2024-09-20 at 15 47 07" src="https://github.com/user-attachments/assets/8523afcf-17d6-4841-a880-68561de0b8e9">
18. Repeat steps 11 and 12
19. As in step 14, the event <span>#</span>1 should be _Consent Default_, but now the table in the _Consent_ tab should have the value _Granted_ in the _On-page Update_ column of row _analytics_storage_ only <img width="1279" alt="Screenshot 2024-09-20 at 15 56 02" src="https://github.com/user-attachments/assets/b4d36f6f-a5f0-4e0a-8504-be4b938841fd">
20. The whole procedure fro step 11 should be repeated once more but instead of clicking the _Accept analytics cookies_ in step 16, you should click on the _Reject analytics cookies_ button
21. In all the remaining steps all values in the _Consent_ table should be _Denied_





[Google Tag Assistant]: https://tagassistant.google.com "Google Tag Assistant" _Consent_ tab should show a similar table where also the
